src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/EndOfServiceReportService.kt

```diff
@@ -64,11 +63,12 @@ class EndOfServiceReportService(
     val draftEndOfServiceReport = getEndOfServiceReport(endOfServiceReportId)
     updateDraftEndOfServiceReportAsSubmitted(draftEndOfServiceReport, submittedByUser)
 
-    referralService.concludeReferral(draftEndOfServiceReport.referral)
-
     endOfServiceReportEventPublisher.endOfServiceReportSubmittedEvent(draftEndOfServiceReport)
 
-    return endOfServiceReportRepository.save(draftEndOfServiceReport)
+    val savedReport = endOfServiceReportRepository.save(draftEndOfServiceReport)
```

Comment:
small thing - do we want to wait to publish the eosr submitted event until after the eosr is saved?

Reply:
Yes I decided to do that on the basis that the code can assume the report is in its final state and there is no chance for it to be changed whilst it's being published.